design/PRODUCER-MIDDLEWARE.md

```diff
@@ -3,6 +3,40 @@
 The producer is responsible for:
 
 - serving the HTTP requests for pages of events
-- output caching of completed pages
+- output/response caching of completed pages
 - collecting of meta data about usage (see persistence design for details)
-- only serve the event feed to authorized consumers
\ No newline at end of file
+- only serve the event feed to authorized consumers
+
+## How
+
+[See caching discussion here](https://github.com/dburriss/event-feed/issues/2).
+ 
+ ```mermaid
+sequenceDiagram
+
+    participant DB as Event Data Store
+    participant S as Consumer Cache
+    participant P as Producer
+    participant C as Consumer
+
+    C ->>+ P: Fetch meta data
+    P -> DB: Retrieve meta data from DB
```

Comment:
Should we include storing metadata-metadata (ie hit-rate)? Or out of scope for now?

Reply:
Good point. Probably worth adding in.